client/app.js

```diff
@@ -0,0 +1,68 @@
+const form = document.querySelector("#new-post-form");
+const btn = document.querySelector("#msg-btn");
+const newPosts = document.querySelector("show-posts");
+
+form.addEventListener("submit", submitPost);
+btn.addEventListener("click", getMessage);
+
+function getAllPosts() {
+  fetch("http://localhost:3000/posts")
+    .then((r) => r.json())
+    .then(appendPosts)
+    .catch(console.warn);
+}
+
+function submitPost(e) {
+  e.preventDefault();
+
+  const postData = {
+    title: e.target.title.value,
+    pseudonym: e.target.pseudonym.value,
+    body: e.target.body.value,
+  };
+
+  const options = {
+    method: "POST",
+    body: JSON.stringify(postData),
+    headers: { "Content-Type": "application/json" },
+  };
+
+  fetch("http://localhost:3000", options)
+    .then((r) => r.json())
+    .then(appendPost)
+    .then(() => e.target.reset())
+    .catch(console.warn);
+
```

Comment:
Here we are posting to the root route - which doesn't have a POST end point. I think you would want this to get routed to your resource routes so "/posts" or "/anonymous" or whatever you decide to call it!